### PR TITLE
[AAQ-577] (with review fixes) Playground feedback

### PR DESCRIPTION
**Pull request by suzinyou**
Reviewer: @sidravi1 
Estimate: 10 min

---

## Ticket

Fixes: #285 #292 

Please see #285! Merged it without pushing review fixes. This PR includes review fixes

### CODE REVIEW

File: admin_app/src/app/playground/components/PlaygroundComponents.tsx [hunk 1]
```diff
@@ -227,8 +233,51 @@ const MessageSkeleton = () => {
...
+  const handleFeedback = (feedbackType: FeedbackSentimentType) => {
+    const { state, setState } = feedbackMapping[feedbackType];
+
+    if (state) {
+      console.log(`Already sent ${feedbackType} feedback`);
+    } else {
+      setState(true);
+      return onFeedbackSend(message as ResponseMessage, feedbackType);
+    }
+  };
```
Comment (suzinyou) [inline admin_app/src/app/playground/components/PlaygroundComponents.tsx:274]: @sidravi1 I have added these. what do you think? (all the console logs should be snack messages! But I have left it in back log :/)

File: admin_app/src/app/playground/components/PlaygroundComponents.tsx [hunk 2]
```diff
@@ -310,17 +360,48 @@ const MessageBox = (message: Message) => {
...
+              marginTop: "5px",
+              display: "flex",
+              justifyContent: "flex-end",
+              alignItems: "center",
+            }}
           >
-            {"<json>"}
-          </Link>
-        )}
+            {message.json.hasOwnProperty("feedback_secret_key") ? (
```
Comment (suzinyou) [inline admin_app/src/app/playground/components/PlaygroundComponents.tsx:372]: Changed to ternary operators
However they are nested!

File: admin_app/src/app/playground/components/PlaygroundComponents.tsx [hunk 3]
```diff
@@ -310,17 +360,48 @@ const MessageBox = (message: Message) => {
...
+                </IconButton>
+                <IconButton
+                  aria-label="thumbs down"
+                  onClick={() => handleFeedback("negative")}
+                  style={feedbackButtonStyle}
+                >
+                  {feedbackMapping.negative.state == true
+                    ? feedbackMapping.negative.onIcon
+                    : feedbackMapping.negative.offIcon}
+                </IconButton>
```
Comment (suzinyou) [inline admin_app/src/app/playground/components/PlaygroundComponents.tsx:391]: Uses `FeedbacMapping`

Just realized this adds another layer of nested ternary operation 😓

File: admin_app/src/app/playground/page.tsx [hunk 4]
```diff
@@ -202,6 +204,38 @@ const Page = () => {
...
+      apiCalls
+        .postResponseFeedback(
+          queryID,
+          feedback_sentiment,
+          feedbackSecretKey,
+          token,
+        )
+        .then((response) => {
+          console.log("Feedback sent successfully: ", response.message);
+        })
```
Comment (suzinyou) [inline admin_app/src/app/playground/page.tsx:231]: I was erroneously checking for `response.status == 200` when we're only getting json response. Stats check is handled by the function in api.ts

File: admin_app/src/app/playground/components/PlaygroundComponents.tsx (hunk 1 above)
Comment (sidravi1) [reply admin_app/src/app/playground/components/PlaygroundComponents.tsx:276]: Looks good!

File: admin_app/src/app/playground/components/PlaygroundComponents.tsx (hunk 3 above)
Comment (sidravi1) [reply admin_app/src/app/playground/components/PlaygroundComponents.tsx:378]: I guess you could iterate over feedbackMapping to create each of these? But don't spend too much time on this.

**Review by sidravi1 — APPROVED** [review]
lgtm! Thanks @suzinyou

**Pull request merged**